### PR TITLE
T13548 validator url

### PR DESCRIPTION
**Pull request by niden**
Hello!

* Type: new feature
* Link to issue: #13548 

**In raising this pull request, I confirm the following:**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/phalcon/cphalcon/blob/master/CONTRIBUTING.md)
- [x] I have checked that another pull request for this purpose does not exist
- [x] I wrote some tests for this PR
- [x] I updated the CHANGELOG

Changed `Phalcon\Validation\Validator\Url` to work with `options` in passed options. (`FILTER_FLAG_PATH_REQUIRED`, `FILTER_FLAG_QUERY_REQUIRED`)

Thanks

### CODE REVIEW

**Comment by niden** [comment]
cc @sergeyklay

File: phalcon/Validation/Validator/Url.zep [hunk 1]
```diff
@@ -57,14 +57,20 @@ class Url extends Validator
...
     {
-        var value, message, label, replacePairs, code;
+        var code, label, message, options, replacePairs, result, value;
 
         let value = validation->getValue(field);
 
-        if !filter_var(value, FILTER_VALIDATE_URL) {
-            let label = this->prepareLabel(validation, field),
+        if fetch options, this->options["flags"] {
+            let result = filter_var(value, FILTER_VALIDATE_URL, ["flags" : options]);
```
Comment (sergeyklay) [inline phalcon/Validation/Validator/Url.zep]: How about 'options' as well ? there is also missed ability to validate like that:
```php
filter_var(
    'http://example.com',
    FILTER_VALIDATE_URL,
    FILTER_FLAG_PATH_REQUIRED // not array
);
```

Comment (niden) [reply phalcon/Validation/Validator/Url.zep]: Fixed

File: CHANGELOG-4.0.md [hunk 2]
```diff
@@ -60,9 +60,10 @@
...
  [#13439](https://github.com/phalcon/cphalcon/issues/13439)
- - Changed `Phalcon\Mvc\Model\Query` to not call `numRows` when selecting data  [#13387](https://github.com/phalcon/cphalcon/issues/13387)
+- Changed `Phalcon\Mvc\Model\Query` to not call `numRows` when selecting data  [#13387](https://github.com/phalcon/cphalcon/issues/13387)
+- Changed `Phalcon\Validation\Validator\Url` to work with passed options. (`FILTER_FLAG_PATH_REQUIRED`, `FILTER_FLAG_QUERY_REQUIRED`) [#13548](https://github.com/phalcon/cphalcon/issues/13548)
 - `Phalcon\Mvc\Model\Query\Builder` will now omit offsets if they are set as 0.
- - `Phalcon\Mvc\Model\Transaction::__construct()` service parameter must be a string or omitted, `Phalcon\Logger\Formatter\Line::__construct()` parameters must be a string or omitted, `Phalcon\Logger\Formatter\Json::__construct()` parameters must be a string or omitted.
+- `Phalcon\Mvc\Model\Transaction::__construct()` service parameter must be a string or omitted, `Phalcon\Logger\Formatter\Line::__construct()` parameters must be a string or omitted, `Phalcon\Logger\Formatter\Json::__construct()` parameters must be a string or omitted.
```
Comment (sergeyklay) [inline CHANGELOG-4.0.md]: Can we have 3 log entries here instead of separating by comma?

**Review by sergeyklay — DISMISSED** [review]

File: CHANGELOG-4.0.md (hunk 2 above)
Comment (niden) [reply CHANGELOG-4.0.md]: Resolved

**Pull request merged**